src/creator.rs

```diff
@@ -54,8 +53,12 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    // initializes the vectors corresponding to the given round (and all between if not there)
+    // updates current_round and initializes the vectors corresponding to the given round (and all between if not there)
     fn init_round(&mut self, round: Round) {
+        if self.current_round < round {
+            self.current_round = round;
+        }
+
         while self.candidates_by_round.len() <= round.into() {
```

Comment:
The loop can replaced with `Vec::resize_with`

Reply:
Done